homework/tai_nguyen/lesson7.2/JS_questions.md

```diff
@@ -0,0 +1,136 @@
+**Question 1:** Write a function printNumbers(from, to) that outputs a number every second, starting from from and ending with to. Make two variants of the solution.
+
+```typescript
+// Using setInterval
+function printNumber(from: number, to: number) {
+  let intervalId: number;
+
+  intervalId = setInterval(() => {
+    if (from <= to) {
+      console.log(from);
+    }
+
+    if (from === to) {
+      clearInterval(intervalId);
+    }
+
+    from++;
+  }, 1 * 1000);
+}
+
+// Using nested setTimeout
+function printNumber(from: number, to: number) {
+  function schedule(x: number) {
+    if (x > to) {
+      return;
+    }
+
+    setTimeout(() => {
+      console.log(x);
+      schedule(x + 1);
+    }, 1 * 1000);
+  }
+
+  schedule(from);
+}
+```
+
+**Question 2:** Use the reduce method in combination with the concat method to “flatten” an array of arrays into a single array that has all the elements of the original arrays.
+
+```typescript
+function flatten(array: any[][]) {
```

Comment:
It doesn't work for containing multiple sublayers. Example test case: `[[1, 2, 3, [1]], [4, 5], [6]]`.

Reply:
LGTM